commit 6e1907e17734d348c2eddcb1d49764f5e871a4ed
Author: Nir Dotan <nird@mellanox.com>
Date:   Fri Aug 3 15:57:43 2018 +0300

    mlxsw: core_acl_flex_actions: Remove redundant counter destruction
    
    [ Upstream commit 7cc6169493990dec488eda0a3f6612729ca25e81 ]
    
    Each tc flower rule uses a hidden count action. As counter resource may
    not be available due to limited HW resources, update _counter_create()
    and _counter_destroy() pair to follow previously introduced symmetric
    error condition handling, add a call to mlxsw_afa_resource_del() as part
    of the counter resource destruction.
    
    Fixes: c18c1e186ba8 ("mlxsw: core: Make counter index allocated inside the action append")
    Signed-off-by: Nir Dotan <nird@mellanox.com>
    Reviewed-by: Petr Machata <petrm@mellanox.com>
    Reviewed-by: Jiri Pirko <jiri@mellanox.com>
    Signed-off-by: Ido Schimmel <idosch@mellanox.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/ethernet/mellanox/mlxsw/core_acl_flex_actions.c b/drivers/net/ethernet/mellanox/mlxsw/core_acl_flex_actions.c
index d664cc0289c2..a54f23f00a5f 100644
--- a/drivers/net/ethernet/mellanox/mlxsw/core_acl_flex_actions.c
+++ b/drivers/net/ethernet/mellanox/mlxsw/core_acl_flex_actions.c
@@ -584,6 +584,7 @@ static void
 mlxsw_afa_counter_destroy(struct mlxsw_afa_block *block,
 			  struct mlxsw_afa_counter *counter)
 {
+	mlxsw_afa_resource_del(&counter->resource);
 	block->afa->ops->counter_index_put(block->afa->ops_priv,
 					   counter->counter_index);
 	kfree(counter);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1011_6e1907e17734d348c2eddcb1d49764f5e871a4ed.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1011_6e1907e17734d348c2eddcb1d49764f5e871a4ed.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
